src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+
+    /**
+     * @return the amount of balls inside the cargo.
+     */
+    public int getCargoCount() {
+        return position.size();
+    }
+
+    /**
+     * Set motor power.
+     *
+     * @param power the power to the motor [-1, 1].
+     */
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    /**
+     * open the "flap" - solenoid
+     */
+    public void openFlap() {
+        flap.set(false);
+    }
+
+    /**
+     * closes the "flap" - solenoid
+     */
+    public void closeFlap() {
+        flap.set(true);
+    }
+
+    /**
+     * toggles the "flap" - solenoid
+     */
+    public void toggle() {
+        flap.toggle();
+    }
+
+    private void updateActualBallPositions() {
+        var colorIntake = getColor();
+        boolean currentPostFlapBeamInput = postFlapBeam.get();
+        SmartDashboard.putString("alliance", colorIntake.name());
+        if (currentPostFlapBeamInput && !lastPostFlapBeamInput && motor.getMotorOutputPercent() > 0) {
+            position.removeFirst();
+        }
+        if (colorIntake != lastSeenColor) {
+            if (motor.getMotorOutputPercent() < 0) {
+                position.removeLast();
+            } else if (motor.getMotorOutputPercent() > 0) {
+                position.add(colorIntake.name());
+            }
+        }
+        lastPostFlapBeamInput = currentPostFlapBeamInput;
+        lastSeenColor = colorIntake;
+        SmartDashboard.putString("lastSeen", lastSeenColor.name());
+    }
+
+    private Queue getQueue() {
+        /*
+        Case #1  queue = [None, None]
+        Case #2  queue = [None, Alliance]
+        Case #3  queue = [Alliance, None]
+        Case #4  queue = [Alliance, Alliance]
+        Case #5  queue = [Alliance, Opponent]
+        Case #6  queue = [None, Opponent]
+        Case #7  queue = [Opponent, None]
+        Case #8  queue = [Opponent, Opponent]
+        Case #9  queue = [Opponent, Alliance]
+         */
+        var last = position.getLast();
+        var first = position.getFirst();
+        var invalid = DriverStation.Alliance.Invalid.name();
+        var alliance = DriverStation.getAlliance().name();
+
+        if (last.equals(invalid)) {
+            if (first.equals(invalid)) {
+                return Queue.NoneNone;
+            } else if (first.equals(alliance)) {
+                return Queue.NoneAlliance;
+            }
+            return Queue.NoneOpponent;
+        } else if (last.equals(alliance)) {
+            if (first.equals(invalid)) {
+                return Queue.AllianceNone;
+            } else if (first.equals(alliance)) {
+                return Queue.AllianceAlliance;
+            }
+            return Queue.AllianceOpponent;
+        }
+        if (first.equals(invalid)) {
+            return Queue.OpponentNone;
+        } else if (first.equals(alliance)) {
+            return Queue.OpponentAlliance;
+        }
+        return Queue.OpponentOpponent;

```

Comment:
The diff doesn't render very nicely but I deleted all the `else if`s and replaced them with `simplify if`s. Do you understand why?
~~also please check if I'm correct it's 02:48 AM~~
```suggestion
        if (last.equals(invalid)) {
            if (first.equals(invalid)) {
                return Queue.NoneNone;
            } if (first.equals(alliance)) {
                return Queue.NoneAlliance;
            }
            return Queue.NoneOpponent;
        }
        if (last.equals(alliance)) {
            if (first.equals(invalid)) {
                return Queue.AllianceNone;
            } if (first.equals(alliance)) {
                return Queue.AllianceAlliance;
            }
            return Queue.AllianceOpponent;
        }
        if (first.equals(invalid)) {
            return Queue.OpponentNone;
        } if (first.equals(alliance)) {
            return Queue.OpponentAlliance;
        }
        return Queue.OpponentOpponent;
```

Reply:
He knows he doesn't need the `else if`, he said the code look better to him with the `else if`